ci-operator/step-registry/aws/deprovision/iam-role/aws-deprovision-iam-role-commands.sh

```diff
@@ -46,10 +50,44 @@ function aws_delete_role()
 }
 
 
-echo "Deleting roles ... "
-aws_delete_role $REGION "$(head -n 1 ${SHARED_DIR}/aws_byo_role_name_master)"
-aws_delete_role $REGION "$(head -n 1 ${SHARED_DIR}/aws_byo_role_name_worker)"
+function delete_all()
+{
+    echo "Deleting roles ... "
+    aws_delete_role $REGION "$(head -n 1 ${SHARED_DIR}/aws_byo_role_name_master)"
+    aws_delete_role $REGION "$(head -n 1 ${SHARED_DIR}/aws_byo_role_name_worker)"
+
+    echo "Deleting policy ... "
+    aws --region $REGION iam delete-policy --policy-arn "$(head -n 1 ${SHARED_DIR}/aws_byo_policy_arn_master)"
+    aws --region $REGION iam delete-policy --policy-arn "$(head -n 1 ${SHARED_DIR}/aws_byo_policy_arn_worker)"
+}
+
+master_profile=$(aws --region $REGION ec2 describe-instances --filters "Name=tag:Name,Values=${INFRA_ID}-master*" | jq -r '.Reservations[].Instances[].IamInstanceProfile.Arn' | sort | uniq | awk -F '/' '{print $2}')
+worker_profile=$(aws --region $REGION ec2 describe-instances --filters "Name=tag:Name,Values=${INFRA_ID}-worker*" | jq -r '.Reservations[].Instances[].IamInstanceProfile.Arn' | sort | uniq | awk -F '/' '{print $2}')
+
+master_role=$(aws --region $REGION iam get-instance-profile --instance-profile-name ${master_profile} | jq -r '.InstanceProfile.Roles[0].Arn' | awk -F '/' '{print $2}')
+worker_role=$(aws --region $REGION iam get-instance-profile --instance-profile-name ${worker_profile} | jq -r '.InstanceProfile.Roles[0].Arn' | awk -F '/' '{print $2}')
+
+master_policy_arn=$(aws --region $REGION iam list-attached-role-policies --role-name ${master_role} | jq -j '.AttachedPolicies[].PolicyArn')
+worker_policy_arn=$(aws --region $REGION iam list-attached-role-policies --role-name ${worker_role} | jq -j '.AttachedPolicies[].PolicyArn')
+
+
+ic_platform_role=$(yq-go r "${CONFIG}" 'platform.aws.defaultMachinePlatform.iamRole')
+ic_control_plane_role=$(yq-go r "${CONFIG}" 'controlPlane.platform.aws.iamRole')
+ic_compute_role=$(yq-go r "${CONFIG}" 'compute[0].platform.aws.iamRole')
+
+if [[ ${ic_platform_role} != "" ]]; then
+    aws --region $REGION iam get-role --role-name ${master_role}
```

Comment:
This is some installer check point?

Reply:
Yes, the last step of OCP-40930, the byo-IAM resources are not deleted by installer.
I will add a comment to make it clear.

Reply:
Shall we directly use the value of `ic_platform_role`, `ic_control_plane_role`, `ic_compute_role` to check the pre-existing role are not deleted by installer?

Reply:
Maybe we can reference the same logic as what mentioned in https://github.com/openshift/release/pull/52689/files#r1668114731